File: node/python/node.py
class Node():

    def __init__(self, value=0, child=None):
        self.value = value
        self.child = child
    
    def get_child(self):
        return self.child.value
    
    
    def __str__(self):
        return (str)(self.value)

def print_till_end(start):
    if start.child == None:
        print(start.value)
        return 
    else:
        print(start.value)
        print_till_end(start.child)


if __name__ == "__main__":
    end = Node()
    mid = Node(5,end)
    start = Node(0,mid)
    print("end ",end)
    print("mid ",mid)
    print("start ",start)
    print("child of start ie mid ", start.get_child())
    print_till_end(start)
    print("end of test")

File: node/cpp/src/node.cpp
#include "Node.h"
#include <iostream>

// template <class T>
Node::Node(){
    value = 0;
    child = NULL;
}

// template <class T>
Node::Node(int value, Node &child){
    this->value = value;
    this->child = &child;
}

// template <class T>
std::ostream & operator<<(std::ostream &o, Node &node){
    o<<node.value;
    return o;
}
int Node::get_child(){
    return child->value;
}